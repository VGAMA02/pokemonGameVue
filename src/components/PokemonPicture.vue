<template>
    <div class="pokemon-container">
        <center><img class="hidden-pokemon" :src="imgSrc" alt="POKEMON"></center>
        <center><img class="fade-in" v-if="showPokemon" :src="imgSrc" alt="POKEMON"></center>
    </div>
</template>

<script>
export default {
    props:{
        pokemonId:{
            type: Number,
            required: true
        },
        showPokemon:{
            type: Boolean,
            required: true,
            default: false
        }
    },
    computed:{
        
        imgSrc(){
            return `https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/${this.pokemonId}.svg`
        }
    }
}
</script>

<style scoped>
.pokemon-container {
    height: 200px;
}
img {
     margin: auto;
    height: 200px;
    width: 200px;
    position: absolute;
   left: 45%;
    user-select: none;
    -moz-user-select: none;
    -ms-user-select: none;
    -webkit-user-drag: none;
    -webkit-user-select: none;
}
.hidden-pokemon {
    filter: brightness(0);
}


</style>