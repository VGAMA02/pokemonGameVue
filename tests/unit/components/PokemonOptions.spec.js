import { shallowMount } from '@vue/test-utils'
import PokemonPicture from '@/components/PokemonOptions'
import {pokemons} from '../mocks/pokemons.mock'
describe('PokemonOptions Component', () => {
    let wrapper
    beforeEach(() => {
         wrapper = shallowMount(PokemonPicture, {
            props:{
                pokemons
            }
        })
    })
    test('debe ded hacer match con el snapshopt', () => {
        /* console.log(wrapper.html()) */
        expect(wrapper.html()).toMatchSnapshot()
    })

    test('debe de mostrar las 4 opciones', () => {
        const liTags = wrapper.findAll('li')
        expect(liTags.length).toBe(4)
        expect(liTags[0].text()).toBe('pikachu') 
        expect(liTags[1].text()).toBe('ivysaur') 
        expect(liTags[2].text()).toBe('venusaur') 
        expect(liTags[3].text()).toBe('charmander') 
    })

    test('debe de emitir "selection con sus respectivos parametros al hacer click"', () => {
        const [li1,li2,li3,li4] = wrapper.findAll('li')
        li1.trigger('click')
        li2.trigger('click')
        li3.trigger('click')
        li4.trigger('click')
        expect(wrapper.emitted('selection').length).toBe(4)
        expect(wrapper.emitted('selection')[0]).toEqual([1])
        expect(wrapper.emitted('selection')[0]).toEqual([2])
        expect(wrapper.emitted('selection')[0]).toEqual([3])
        expect(wrapper.emitted('selection')[0]).toEqual([4])




    })
    
})