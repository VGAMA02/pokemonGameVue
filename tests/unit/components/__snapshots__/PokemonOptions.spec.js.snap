// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonOptions Component debe ded hacer match con el snapshopt 1`] = `
<div class="options-container">
  <ul>
    <li>pikachu</li>
    <li>ivysaur</li>
    <li>venusaur</li>
    <li>charmander</li>
  </ul>
</div>
`;
